FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.JobManagement/AzureStorageClientFactory.cs

```diff
@@ -73,6 +76,11 @@ public TableClient CreateTableClient()
                 return new TableClient(_tableUrl, _tableName);
             }
 
+            if (!string.IsNullOrEmpty(_internalConnectionString))
```

Comment:
merge these two if statements, as they are all create `TableClient` by connection string

Reply:
After offline discussion, we keep it unchanged here.